apps/frontend/src/app/5_pages/AavePage/components/BorrowAssetsList/components/BorrowForm/BorrowForm.tsx

```diff
@@ -68,6 +68,14 @@ export const BorrowForm: FC<BorrowFormProps> = ({ asset, onComplete }) => {
     [summary.reserves, borrowAsset],
   );
 
+  useEffect(() => {
+    // if borrow size is greater than available to borrow, set borrow amount to available to borrow
+    // this is to prevent users from borrowing more than they can even if maximum gets updated
+    if (borrowSize.gt(borrowReserve?.availableToBorrow ?? 0)) {
+      setBorrowAmount(borrowReserve?.availableToBorrow.toString() ?? '0');
+    }
+  }, [borrowReserve?.availableToBorrow, borrowSize, setBorrowAmount]);
```

Comment:
Couple of suggestions:

1. I dont think you need to include `setBorrowAmount` as a dependency here. Probably doesnt hurt, but the hook doesnt really depend on it.

2. Also, maybe extract the condition for the if statement in a constant like `isSomeCondition`, just to make this easier to read.

Reply:
Actually I wouldn't use an effect and state to set just an string. A memo like borrowUsdAmount or nothing would be fine!

Reply:
Hey guys thank you for the comments! So, `setBorrowAmount` is mostly to keep the linter happy, they have husky connected and so I cannot lawfully commit without it. 
<img width="786" alt="image" src="https://github.com/user-attachments/assets/8b987ffb-8c9f-454e-97fc-b6666dbec949">
The reason for the useEffect is because I'm using `useDecimalAmountInput` which connects user input with a Decimal (Bignumber) version so to reflect changes for user I think would be necessary )
Agree with the constant for readability, I'll push rn )